src/main/webapp/app/file-upload-submission/file-upload-submission.component.ts

```diff
@@ -0,0 +1,219 @@
+
+        if (confirmSubmit) {
+            this.submission!.submitted = true;
+            this.fileUploadSubmissionService.update(this.submission!, this.fileUploadExercise.id).subscribe(
+                response => {
+                    this.submission = response.body!;
+                    this.result = this.submission.result;
+                    this.setSubmittedFile();
+                    if (this.isActive) {
+                        this.jhiAlertService.success('artemisApp.fileUploadExercise.submitSuccessful');
+                    } else {
+                        this.jhiAlertService.warning('artemisApp.fileUploadExercise.submitDeadlineMissed');
+                    }
+                },
+                err => {
+                    this.jhiAlertService.error('artemisApp.fileUploadExercise.error');
+                    this.submission!.submitted = false;
+                },
+            );
+        }
+    }
+
+    /**
+     * Sets file submission for exercise
+     * @param $event {object} Event object which contains the uploaded file
+     */
+    setFileSubmissionForExercise($event: any): void {
+        if ($event.target.files.length) {
+            this.erroredFile = null;
+            const fileList: FileList = $event.target.files;
+            const submissionFile = fileList[0];
+            const allowedFileExtensions = this.fileUploadExercise.filePattern.split(',');
+            if (allowedFileExtensions.some(extension => submissionFile.name.toLowerCase().endsWith(extension))) {
+                this.submissionFile = submissionFile;
+            } else {
+                this.jhiAlertService.error('artemisApp.fileUploadSubmission.fileExtensionError', null, undefined);
+            }
+        }
+    }
+
+    private onError(error: HttpErrorResponse) {
+        this.jhiAlertService.error(error.message, null, undefined);
+    }
+
+    private setSubmittedFile() {
+        const filePath = this.submission!.filePath!.split('/');
+        this.submittedFileName = filePath[filePath.length - 1];
+        const fileName = this.submittedFileName.split('.');
+        this.submittedFileExtension = fileName[fileName.length - 1];
+    }
+
+    /**
+     * Navigates to previous location
+     */
+    previous() {
+        this.location.back();
+    }
+
+    downloadFile(filePath: string) {
+        this.fileService.downloadAttachment(filePath);
+    }
+
+    /**
+     * Hides more feedback form and shows complaint form
+     */
+    toggleComplaintForm() {
+        this.showRequestMoreFeedbackForm = false;
+        this.showComplaintForm = !this.showComplaintForm;
+    }
+
+    /**
+     * Hides complaint form and shows more feedback form
+     */
+    toggleRequestMoreFeedbackForm() {
+        this.showComplaintForm = false;
+        this.showRequestMoreFeedbackForm = !this.showRequestMoreFeedbackForm;
+    }
+
+    /**
+     * Returns false if user selected a file, but didn't submit the exercise, false otherwise.
```

Comment:
The documentation doesn't make sense atm, because you say that either case `false` is returned. Also I am a bit confused: do we need `Observable<boolean>` as a return type if we only return `boolean`?

Reply:
Good catch. That was a typo and returning `boolean` is cleaner. Changed.